src/MAUI/Maui.Samples/Samples/GraphicsOverlay/SketchOnMap/SketchOnMap.xaml.cs

```diff
@@ -213,10 +216,85 @@ private async void EditButtonClick(object sender, EventArgs e)
             }
         }
 
-        private void ShowDrawTools(object sender, EventArgs e)
+        private async void ShowDrawTools(object sender, EventArgs e)
         {
-            // Show the grid that contains the sketch mode, draw, and edit controls.
-            DrawToolsGrid.IsVisible = true;
+            try
+            {
+                // Update UI.
+                SelectTool(sender as Button);
+
+                // Show draw tools and wait for the user to select a mode.
+                string choice = await Application.Current.MainPage.DisplayActionSheet("Set draw mode:", "Cancel", null, _sketchModes);
+
+                // Set the selected index for the draw mode.
+                if (_sketchModes.Contains(choice))
+                {
+                    _drawModeIndex = Array.IndexOf(_sketchModes, choice);
+                    StartSketch(sender, e);
+                }
+                else
+                {
+                    throw new OperationCanceledException();
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                UnselectTool(sender, e);
```

Comment:
We throw this exception only to catch it ourselves. There are two changes we can make here.
1. Moving the unselect tool call to the else section.
2. Add a `catch(Exception)` for any exception that might be raised by the awaited call to the action sheet. `async void` is dangerous territory, and we surround async calls in a try/catch to prevent un caught exceptions crashing the program.